Add support for loading config values on object initialization not only class creation (#22)
jmcs/ecological: README.rst

.. code-block:: python

    class Configuration(ecological.Config):
        port: int
        debug: bool
        log_level: str = "INFO"

And then set the environment variables ``PORT``, ``DEBUG`` and ``LOG_LEVEL``. ``Ecological`` will automatically set the
class properties from the environment variables with the same (but upper cased) name.

By default the values are set at the class definition type and assigned to the class itself (i.e. the class doesn't need to be
instantiated). If needed this behavior can be changed (see the next section).

Autoloading
=============

> It is possible to defer/disable autoloading (setting) of variable values by specifyfing ``autoload`` option on class definition.

Reviewer: ```suggestion
It is possible to defer/disable autoloading (setting) of variable values by specifying the ``autoload`` option on class definition.
```

On class creation (default)
---------------------------
When no option is provided values are loaded immediately on class creation and assigned to class attributes:

.. code-block:: python

>     class Configuration(ecological.Config, autoload=ecological.Autoload.CLASS):

Reviewer: If it's the default, it would perhaps be clearer to give a first example without the option at all? Basically for illustrative purposes.

        port: int
    # Values already read and set at this point.
    # assert Configuration.port == <value-of-PORT-env-var>

Never
------

> When this option is chosen, no autoloading happens. In order to set variable values ``Config.load`` method needs to be called explicitly:

Reviewer: ```suggestion
When this option is chosen, no autoloading happens. In order to set variable values, the ``Config.load`` method needs to be called explicitly:
```

.. code-block:: python

>     class Configuration(ecological.Config, autoload=ecological.Autoload.CLASS):

Reviewer: This section is dedicated to the `Never` option but `CLASS` is used instead.
```suggestion
    class Configuration(ecological.Config, autoload=ecological.Autoload.NEVER):
```

        port: int
    # Values not set at this point.
    # Accessing Configuration.port would throw AttributeError.

    Configuration.load()
    # Values read and set at this point.
    # assert Configuration.port == <value-of-PORT-env-var>

On object instance initialization
----------------------------------

> If it is preferred to load and store attribute values on the object instance instead class itself, ``Autoload.OBJECT`` strategy can be used:

Reviewer: ```suggestion
If it is preferred to load and store attribute values on the object instance instead of the class itself, the ``Autoload.OBJECT`` strategy can be used:
```
